utilities/test_suite/HIP/runTests.py

```diff
@@ -291,9 +507,190 @@ def rpp_test_suite_parser_and_validator():
             # Closing log file
             f.close()
     elif (testType == 1 and profilingOption == "YES"):
-        subprocess.call(["./testAllScript.sh", srcPath1, srcPath2, str(testType), str(numRuns), "1", str(qaMode), str(decoderType), str(preserveOutput), str(batchSize), " ".join(caseList)])  # nosec
         NEW_FUNC_GROUP_LIST = [0, 15, 20, 29, 36, 40, 42, 49, 56, 65, 69]
 
+        for case in caseList:
+            if int(case) < 0 or int(case) > 84:
+                print(f"Invalid case number {case}. Case number must be in the range of 0 to 84!")
+                continue
+            for layout in range(3):
+                if layout == 0:
+                    directory_name_generator(qaMode, "hip", "pkd3", case, dstPath)
+                    log_file_layout = "pkd3"
+                elif layout == 1:
+                    directory_name_generator(qaMode, "hip", "pln3", case, dstPath)
+                    log_file_layout = "pln3"
+                elif layout == 2:
+                    directory_name_generator(qaMode, "hip", "pln1", case, dstPath)
+                    log_file_layout = "pln1"
+
+                print("\n\n\n\n")
+                print("--------------------------------")
+                print("Running a New Functionality...")
+                print("--------------------------------")
+                
+                for bitDepth in range(7):
+                    print("\n\n\nRunning New Bit Depth...\n-------------------------\n\n")
+                    
+                    for outputFormatToggle in range(2):
+                        # There is no layout toggle for PLN1 case, so skip this case
+                        if layout == 2 and outputFormatToggle == 1:
+                            continue
+
+                        if case == 40 or case == 41 or case == 49:
```

Comment:
Would suggest a little more modularization wherever you can - for example the following lines are common everywhere with minor differences. 
```
if not os.path.isdir
    os.mkdir
with open
    print()
    process = subprocess.Popen()
    while True:
        ...
```
Or in any other repeated code segments

Reply:
Done